### PR TITLE
Compile trace memory

### PR DESCRIPTION
**Pull request by Juan-M-V**
# TITLE

## Description

Compile trace and memory at the same time.
Previously the memory and trace file where computed with cairo-run in separate runs:
`cairo-run --program program_name.json --memory_file program_name.memory`
`cairo-run --program program_name.json --trace_file program_name.trace`

This has now been merged into one run:
`cairo-run --program program_name.json --memory_file program_name.memory --trace_file program_name.trace`
Saving time on the memory and trace comparison.
## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Comment by entropidelic** [comment]
Please add a description so that the changes made are a bit clearer

File: src/cairo_run.rs [hunk 1]
```diff
@@ -217,8 +217,8 @@ mod tests {
     #[test]
     fn write_binary_trace_file() {
         let program_path = Path::new("cairo_programs/struct.json");
-        let expected_trace_path = Path::new("cairo_programs/struct.trace");
-        let cairo_rs_trace_path = Path::new("cairo_programs/struct_rs.trace");
+        let expected_trace_path = Path::new("cairo_programs/trace_memory/cairo_trace_struct");
+        let cairo_rs_trace_path = Path::new("cairo_programs/trace_memory/struct_cleopatra.trace");
```
Comment (fmoletta) [inline src/cairo_run.rs:221]: Please change the name of the files as Cleopatra is no longer the official name

File: src/cairo_run.rs [hunk 2]
```diff
@@ -241,8 +241,8 @@ mod tests {
     #[test]
     fn write_binary_memory_file() {
         let program_path = Path::new("cairo_programs/struct.json");
-        let expected_memory_path = Path::new("cairo_programs/struct.memory");
-        let cairo_rs_memory_path = Path::new("cairo_programs/struct_rs.memory");
+        let expected_memory_path = Path::new("cairo_programs/trace_memory/cairo_memory_struct");
+        let cairo_rs_memory_path = Path::new("cairo_programs/trace_memory/struct_cleopatra.memory");
```
Comment (fmoletta) [inline src/cairo_run.rs:245]: Same here

Comment (Juan-M-V) [reply src/cairo_run.rs]: Resolved: https://github.com/lambdaclass/cairo-rs/pull/386/commits/76c7b1433769189d7d77dda9d368b1fd18eb384c

File: src/cairo_run.rs (hunk 1 above)
Comment (Juan-M-V) [reply src/cairo_run.rs]: Resolved: https://github.com/lambdaclass/cairo-rs/pull/386/commits/76c7b1433769189d7d77dda9d368b1fd18eb384c

**Review by fmoletta — APPROVED** [review]
Nice Work!

**Review by entropidelic — APPROVED** [review]

**Pull request merged**